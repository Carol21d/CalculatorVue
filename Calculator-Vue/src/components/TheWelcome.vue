<script>
export default {
  data() {
    return {
      result: '',
      buttons: [
        { id: 'multiplication', label: '*' },
        { id: 'division', label: '/' },
        { id: 'addition', label: '+' },
        { id: 'subtraction', label: '-' },
        { id: 'coma', label: '.' },
        { id: 'equals', label: '=' },
        { id: 'cleanScreen', label: 'C' },
        { id: 'zero', label: '0' },
        { id: 'one', label: '1' },
        { id: 'two', label: '2' },
        { id: 'three', label: '3' },
        { id: 'four', label: '4' },
        { id: 'five', label: '5' },
        { id: 'six', label: '6' },
        { id: 'seven', label: '7' },
        { id: 'eight', label: '8' },
        { id: 'nine', label: '9' }
      ]
    }
  },

  methods: {
    operationChoose(button) {
      switch (button.id) {
        case 'cleanScreen':
          this.result = '';
          break;
        case 'equals':
          this.result = eval(this.result);
          break;
        case 'multiplication':
          this.result += ' ' + button.label + ' ';
          break;
        case 'divide':
          this.result += ' ' + button.label + ' ';
          break;
        case 'addition':
          this.result += ' ' + button.label + ' ';
          break;
        case 'subtract':
          this.result += ' ' + button.label + ' ';
          break;
        default:
          this.result += button.label;
      }
    }
  },
}
</script>
<template>
  <div class="wrapper">
    <h1 class="wrapper-title">Calculator</h1>
    <div class="wrapper-display">
      <input type="text" v-model="result" placeholder="0" readonly />
    </div>
    <div class="wrapper-buttons">
      <button v-for="button in buttons" :key="button.id" @click="operationChoose(button)">{{ button.label }}</button>
    </div>
  </div>
</template>
                         
    
    
                













