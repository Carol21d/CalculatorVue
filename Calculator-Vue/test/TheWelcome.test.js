import { describe, it, expect, test } from 'vitest'
import { mount } from '@vue/test-utils'
import TheWelcome from '../src/components/TheWelcome.vue'


describe.skip('TheWelcome', () => {
  test('wrapper contains a h1 with name Calculator', () => {
    const wrapper = mount(TheWelcome)
 
    expect(wrapper.find('h1').text()).toBe('Calculator')
  })
})

describe.skip('TheWelcome', () =>{
  it('contains a class named wrapper-tittle ', () =>{
    const wrapperClass = mount(TheWelcome)
    expect(wrapperClass.find('.wrapper-title').exists())
  
})
})

describe.skip('TheWelcome', () =>{
  test(' button contains button.id', () =>{
    const wrapperButton = mount(TheWelcome)
    expect(wrapperButton.find('button').attributes('#button.id'))
  })
})