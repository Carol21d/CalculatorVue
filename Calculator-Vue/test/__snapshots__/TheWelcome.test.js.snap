// Vitest Snapshot v1

exports[`TheWelcome > renders properly 1`] = `
"<div class=\\"wrapper\\">
  <div class=\\"wrapper-display\\"><input type=\\"text\\" placeholder=\\"0\\" readonly=\\"\\"></div>
  <div class=\\"wrapper-buttons\\"><button>*</button><button>/</button><button>+</button><button>-</button><button>.</button><button>=</button><button>C</button><button>0</button><button>1</button><button>2</button><button>3</button><button>4</button><button>5</button><button>6</button><button>7</button><button>8</button><button>9</button></div>
</div>"
`;
